src/esp/gfx/replay/Recorder.h

```diff
@@ -126,10 +126,23 @@ class Recorder {
                                  bool usePrettyWriter = false);
 
   /**
-   * @brief write saved keyframes to string.
+   * @brief write saved keyframes to string. '{"keyframes": [{...},{...},...]}'
    */
   std::string writeSavedKeyframesToString();
 
+  /**
+   * @brief write saved keyframes as individual strings ['{"keyframe": ...}',
+   * '{"keyframe": ...}', ...]
+   * @param incremental whether future saved keyframes will be incremental
+   *
+   * Set incremental to true if you are using keyframes incrementally, e.g.
+   * repeated calls to this function and feeding them to a renderer. Set to
+   * false if you want the next call to writeSavedKeyframes* to be "standalone",
+   * i.e. the next frame will properly initialize all scene state and not rely
+   * on previous keyframes.
+   */
+  std::vector<std::string> writeSavedKeyframesToStringArray(bool incremental);
```

Comment:
Boolean arguments are an API smell -- `writeSavedKeyframesToStringArray(false)` could imply "not write", "not saved", "not array" or any other meaning until you actually check the API documentation.

For consistency with `writeSavedKeyframesToString()` i think the easiest would be adding a new `writeIncrementalSavedKeyframesToStringArray()` variant that doesn't call the `consolidate*()` inside.